src/sdk/namespace_fs.js

```diff
@@ -1050,6 +1063,13 @@ class NamespaceFS {
                 upload_params.digest = await buffers_pool_sem.surround_count(
                     config.NSFS_BUF_SIZE, async () => this._upload_stream(upload_params));
             }
+
+            if (params.storage_class) {
+                await upload_params.target_file.replacexattr(fs_context, {
```

Comment:
we prefer setting all the xattrs together on finish_upload() if possible

Reply:
Actually I did put this inside `_finish_upload` but later on pulled it out. I think I did it to ensure that this isn't executed in `upload_multipart` but I should be able to easily avoid that case (as no storage class is passed to `upload_multipart`). I will test pushing it back inside `_finish_upload`.

Reply:
cool, you can check for (!part_upload) inside finish_upload()